api/metric/must.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/api/core"
+)
+
+// MeterMust is a variation on Meter that panics when any instrument
+// constructor encounters an error.
+type MeterMust interface {
```

Comment:
Does it need to be an interface instead of a concrete type?

Reply:
No, it doesn't. I had some reasons in the back of my mind why it should, but I'll do this.